Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -152,46 +152,44 @@ internal class ModuleMetadataEmitter(
 
         override fun visitFunction(element: FunctionStub, data: VisitingContext) =
                 data.withMappingExtensions {
-                    if (bridgeBuilderResult.nativeBridges.isSupported(element)) {
-                        KmFunction(element.flags, element.name).also { km ->
-                            km.receiverParameterType = element.receiver?.type?.map()
-                            element.typeParameters.mapTo(km.typeParameters) { it.map() }
-                            element.parameters.mapTo(km.valueParameters) { it.map() }
-                            element.annotations.mapTo(km.annotations) { it.map() }
-                            km.returnType = element.returnType.map()
-                            km.uniqId = data.uniqIds.uniqIdForFunction(element)
-                        }
+                    val annotations = if (bridgeBuilderResult.nativeBridges.isSupported(element)) {
+                        element.annotations
                     } else {
-                        null
+                        listOf(AnnotationStub.Deprecated.unableToImport)
+                    }
+                    KmFunction(element.flags, element.name).also { km ->
```

Comment:
Not clear for me: text emitter rendered "unsupported" declarations as non-external ones.
Metadata emitter doesn't seem to do so. Is this true? If yes, does our infrastructure detect such a difference?

Reply:
Nice catch: test infrastructure doesn't check flags of `KmFunction`.